### PR TITLE
Implement Edit Tag Command and Delete Tag Command

### PR DESCRIPTION
**Pull request by Li-Zizhen**
Implement Edit Tag Command and Delete Tag Command. They are well-tested.

### CODE REVIEW

**Review by ValenciaLim — COMMENTED** [review]
Great test cases added. Good that you checked for duplicate tags before implementing edit. (by checking for disjoint set).

**Review by ValenciaLim — COMMENTED** [review]
Noted that you used null value for your studentToEdit variable in EditTagCommand and DeleteTagCommand. Recommended to not use null value.

**Comment by Li-Zizhen** [comment]
> Noted that you used null value for your studentToEdit variable in EditTagCommand and DeleteTagCommand. Recommended to not use null value.

Sorry, may I know the reason for that? Thank you.

File: src/main/java/seedu/edulink/logic/commands/DeleteTagCommand.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        requireAllNonNull(personToEditId, tags);
+
+        this.personToEditId = personToEditId;
+        this.tags = tags;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Student> lastShownList = model.getFilteredPersonList();
+        Student studentToEdit = null;
```
Comment (dedsecrattle) [inline src/main/java/seedu/edulink/logic/commands/DeleteTagCommand.java:50]: Can we avoid the  use of `null` here and instead try using `Optional` or just manipulate the `lastShownList` using Streams

**Review by dedsecrattle — CHANGES_REQUESTED** [review]
Please avoid the uses of null the code

File: src/main/java/seedu/edulink/logic/commands/EditTagCommand.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        this.personToEditId = personToEditId;
+        this.tagToEdit = tagToEdit;
+        this.resultingTag = resultingTag;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Student> lastShownList = model.getFilteredPersonList();
+        Student studentToEdit = null;
```
Comment (dedsecrattle) [inline src/main/java/seedu/edulink/logic/commands/EditTagCommand.java:57]: Similarly here null can be avoided

File: src/main/java/seedu/edulink/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -85,6 +87,12 @@ public Command parseCommand(String userInput) throws ParseException {
             return new UndoCommand();
         case TagCommand.COMMAND_WORD:
             return new TagCommandParser().parse(arguments);
+
+        case DeleteTagCommand.COMMAND_WORD:
+            return new DeleteTagCommandParser().parse(arguments);
```
Comment (dedsecrattle) [inline src/main/java/seedu/edulink/logic/parser/AddressBookParser.java:92]: edit the AddressBookParserTest file to test these lines of code as well

**Review by dedsecrattle — CHANGES_REQUESTED** [review]
Following changes request - 

1. Avoid uses of Null
2. Test cases for Newly added Parser Class

File: src/main/java/seedu/edulink/logic/parser/AddressBookParser.java (hunk 3 above)
Comment (Li-Zizhen) [reply src/main/java/seedu/edulink/logic/parser/AddressBookParser.java:96]: Thank you for your comments. Issue resolved.

File: src/main/java/seedu/edulink/logic/commands/DeleteTagCommand.java (hunk 1 above)
Comment (Li-Zizhen) [reply src/main/java/seedu/edulink/logic/commands/DeleteTagCommand.java]: Thank you, issue resovled.

**Review by dedsecrattle — APPROVED** [review]
Looks good to me

**Pull request merged**